src/System.Text.Json/tests/Utf8JsonReaderTests.cs

```diff
@@ -1512,6 +1503,89 @@ public static void InvalidJsonWithCommentsSingleSegment(string jsonString, int e
             }
         }
 
+        [Fact]
+        public static void EmptyJsonIsInvalid()
+        {
+            var dataUtf8 = ReadOnlySpan<byte>.Empty;
+            var json = new Utf8JsonReader(dataUtf8, isFinalBlock: true, state: default);
+
+            try
+            {
+                while (json.Read())
+                    ;
+                Assert.True(false, "Expected JsonReaderException was not thrown with single-segment data.");
+            }
+            catch (JsonReaderException ex)
+            {
+                Assert.Equal(0, ex.LineNumber);
+                Assert.Equal(0, ex.BytePositionInLine);
+            }
+        }
+
+        [Fact]
+        public static void EmptyJsonWithinSequenceIsInvalid()
+        {
+            var dataUtf8 = ReadOnlySpan<byte>.Empty;
+            ReadOnlySequence<byte> sequence = JsonTestHelper.GetSequence(new byte[0], 1);
+            var json = new Utf8JsonReader(dataUtf8, isFinalBlock: true, state: default);
+
+            try
+            {
+                while (json.Read())
+                    ;
+                Assert.True(false, "Expected JsonReaderException was not thrown with single-segment data.");
+            }
+            catch (JsonReaderException ex)
+            {
+                Assert.Equal(0, ex.LineNumber);
+                Assert.Equal(0, ex.BytePositionInLine);
+            }
+        }
+
+        [Fact]
+        public static void JsonContainingOnlyWhitespaceIsInvalid()
+        {
+            var dataUtf8 = new byte[] { 0x20 };
+            var json = new Utf8JsonReader(dataUtf8, isFinalBlock: true, state: default);
+
+            try
+            {
+                while (json.Read())
+                    ;
+                Assert.True(false, "Expected JsonReaderException was not thrown with single-segment data.");
+            }
+            catch (JsonReaderException ex)
+            {
+                Assert.Equal(0, ex.LineNumber);
+                Assert.Equal(1, ex.BytePositionInLine);
+            }
+        }
+
+        [Theory]
+        [InlineData("//", 2)]
+        [InlineData("//\n", 3)]
+        [InlineData("/**/", 4)]
+        [InlineData("/*/*/", 5)]
+        [InlineData("// just a comment", 17)]
+        [InlineData(" /* comment and whitespace */ ", 30)]
+        public static void JsonContainingOnlyCommentsIsInvalid(string jsonString, int expectedConsumed)
```

Comment:
I think the `ex.BytePositionInLine` for `//\n` would be 0 and `ex.LineNumber` would be 1.

We should also add a test for when `CommentHandling` is set to `JsonCommentHandling.Allow`.

Reply:
It seems there are already tests with `JsonCommentHandling.Allow`